src/Main.py

```diff
@@ -0,0 +1,636 @@
+        self.window.set_application(self)
+        self.window.present()
+
+    def on_button_mainpage_clicked(self, widget):
+        self.deck.set_visible_child(self.page_main)
+
+    def on_button_show_log_clicked(self, widget):
+        self.box_vte.set_visible(True)
+        self.status_page.set_visible(False)
+        self.btn_go_mainpage.set_visible(False)
+        self.btn_close_logs.set_visible(True)
+
+    def on_button_close_logs_clicked(self, widget):
+        self.box_vte.set_visible(False)
+        self.status_page.set_visible(True)
+        self.btn_go_mainpage.set_visible(True)
+        self.btn_close_logs.set_visible(False)
+
+    def on_row_advanced_options_activated(self, widget):
+        self.deck.set_visible_child(self.page_advanced)
+
+    """
+        row functions is seperated to at least 2 subfunctions
+        pre() and post()
+        pre() function will be called first and it will prepare the page for the user input or execute the command then set the post_command to the post() function
+        post_command is a function that will be called after the vte command is executed (see vte_cb() function)
+        post() function will be called after the command is executed, it will update the status page and do the necessary actions
+    """
+    """
+        get_user, get_rootfs, get_mbr functions are used to get the user, rootfs and mbr (will try to get automatically if there is only one)
+        if there is no user, rootfs or mbr, it will show an error message
+        if there is only one user, rootfs or mbr, it will set self.user, self.rootfs or self.mbr to it
+        if there is more than one user, rootfs or mbr, the function will return None but it will create a listbox page for the user to select one of them
+        after the user selects one of them, the get_user, get_rootfs or get_mbr function will call the row function again (see self.pending_func)
+    """
+    def on_row_reinstall_grub_activated(self, widget):
+        def pre():
+            self.pending_func = pre
+            self.deck.set_visible_child(self.page_loading)
+            for child in self.carousel_questions.get_children():
+               self.carousel_questions.remove(child)
+            if self.get_rootfs(widget) == None:
+                return
+            if self.get_mbr(widget) == None:
+                return
+            if os.path.exists("/sys/firmware/efi/efivars") and self.get_clearEfivars(widget) == None: 
+                return
+            self.update_status_page(_("Reinstalling GRUB Bootloader"), "content-loading-symbolic", _("We're reinstalling the GRUB boot loader to ensure your system can start up properly. This process may take a few moments. Once complete, your computer should boot into Pardus as usual."), False, False)
+            self.post_command = post
+            if self.rootfs.root_subvol == None:
+                self.vte_command("env disk={} mbr={} pardus-reinstall".format(self.rootfs.name, self.mbr))
```

Comment:
trigger grub-reinstall instead of pardus-reinstall

Reply:
done